tests/hypervisor/test_esx.py

```diff
@@ -93,6 +93,226 @@ def test_hypervisor_id(self, virtwho, function_hypervisor, hypervisor_data, glob
+            3. Configure filter_hosts='*', run the virt-who service.
+            4. Configure filter_hosts='', run the virt-who service.
+            5. Delete filter_hosts, Configure exclude_hosts=host_uuid, run the virt-who service.
+            6. Configure exclude_hosts='*', run the virt-who service.
+            7. Configure exclude_hosts='', run the virt-who service.
+            8. run the above cases with hypervisor_id =hostname and hwuuid
+            9. run virt-who with filter_hosts=[host_uuid] and exclude_hosts=[host_uuid]
+            10. run virt-who with filter_hosts=* and exclude_hosts=[host_uuid]
+            11. run virt-who with exclude_hosts= and filter_hosts=[host_uuid]
+
+        :expectedresults:
+            2. Succeeded to run the virt-who, can find the log message "hypervisorId": "{host_uuid}"
+            3. Succeeded to run the virt-who, can find the log message "hypervisorId": "{host_uuid}"
+            4. Succeeded to run the virt-who, cannot find the log message "hypervisorId": "{host_uuid}"
+            5. Succeeded to run the virt-who, cannot find the log message "hypervisorId": "{host_uuid}"
+            6. Succeeded to run the virt-who, cannot find the log message "hypervisorId": "{host_uuid}"
+            7. Succeeded to run the virt-who, can find the log message "hypervisorId": "{host_uuid}"
+            8. The result is the same as above
+            9. Succeeded to run the virt-who, cannot find the log message "hypervisorId": "{host_uuid}"
+            10. Succeeded to run the virt-who, cannot find the log message "hypervisorId": "{host_uuid}"
+            11. Succeeded to run the virt-who, can find the log message "hypervisorId": "{host_uuid}"
+
+        """
+        hypervisor_ids = ['hostname', 'uuid', 'hwuuid']
+        for hypervisor_id in hypervisor_ids:
+            function_hypervisor.update('hypervisor_id', hypervisor_id)
+            hypervisor_id_data = hypervisor_data[f'hypervisor_{hypervisor_id}']
+
+            # run virt-who with filter_hosts option
+            for filter_hosts in [hypervisor_id_data, '*', '']:
+                function_hypervisor.update('filter_hosts', filter_hosts)
+                result = virtwho.run_service()
+                if filter_hosts == '':
+                    assert (result['error'] == 0
+                            and result['send'] == 1
+                            and result['thread'] == 1
+                            and hypervisor_id_data not in str(result['mappings']))
+                else:
+                    assert (result['error'] == 0
+                            and result['send'] == 1
+                            and result['thread'] == 1
+                            and hypervisor_id_data in str(result['mappings']))
+
+            function_hypervisor.delete('filter_hosts')
+
+            # run virt-who with exclude_hosts option
+            for exclude_hosts in [hypervisor_id_data, '*', '']:
```

Comment:
Same suggestion to only test the `hypervisor_id_data` for exclude_hosts in tier1